src/jit/hwintrinsicxarch.cpp

```diff
@@ -295,6 +304,163 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types argType, CORINFO_CLASS_HANDLE
     return arg;
 }
 
+//------------------------------------------------------------------------
+// immUpperBoundOfHWIntrinsic: get the max imm-value of non-full-range IMM intrinsic
+//
+// Arguments:
+//    intrinsic  -- intrinsic ID
+//    baseType   -- base type of the Vector128/256<T>
+//
+// Return Value:
+//     the max imm-value of non-full-range IMM intrinsic
+//
+int Compiler::immUpperBoundOfHWIntrinsic(NamedIntrinsic intrinsic, var_types baseType)
```

Comment:
I just don't understand why people insist in adding more stuff to `Compiler`...

Reply:
Hmm.. these functions are needed by multiple passes (e.g., importer, codegen, lowering, etc). Is there other better approach to organize these functions?

Reply:
One reasonable option seems to be for these functions to be static members of `HWIntrinsicInfo`. Haven't really checked if this is actually possible or not. Another option might be global functions, similar to `varType` functions for example. Yet another option would be a `HWIntrinsicTable` class that basically acts as an abstraction for `hwIntrinsicInfoArray`.

That said, I'm not sure if it's worth bothering with this ATM.